tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/MetaSyntaxGenerator.java

```diff
@@ -0,0 +1,23 @@
+package fi.helsinki.cs.tmc.langs.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MetaSyntaxGenerator {
+    
+    public static List<MetaSyntax> listSyntaxes(String fileType) {
+        List<MetaSyntax> list = new ArrayList<>();
+        if (fileType.matches("^$|java|c|cpp|h|hpp|js")) {
+            list.add(new MetaSyntax("\\/\\/", ""));          //
+            list.add(new MetaSyntax("\\/\\*+", "\\*+\\/"));  /* */
+        }
+        if (fileType.matches("^$|xml|http|html")) {
+            list.add(new MetaSyntax("<!--", "-->"));
+        }
+        
+        // If filetype is not recognized, list all syntaxes
+        if (list.isEmpty() && !fileType.isEmpty()) return listSyntaxes("");
```

Comment:
Use explicit blocks even for one-liners.

I was also wondering why this behavior is preferable to returning an empty list. At least for me it breaks the [Principle of Least Astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment): "we don't know what to do so we do everything" vs. "we don't know what to do so we do nothing".